pkg/registration/activate.go

```diff
@@ -21,16 +25,104 @@ type Metadata struct {
 	ObsoletedName string `json:"obsoleted_service_name"`
 }
 
+type activateRequest struct {
+	Identifier string `json:"identifier"`
+	Version    string `json:"version"`
+	Arch       string `json:"arch"`
+	Regcode    string `json:"token,omitempty"`
+}
+
+type activateResponse struct {
+	ID            int     `json:"id"`
+	URL           string  `json:"url"`
+	Name          string  `json:"name"`
+	ObsoletedName string  `json:"obsoleted_service_name"`
+	Product       Product `json:"product"`
+}
+
 // Activate a product by pairing an authorized connection (which contains the
 // system at hand), plus the "triplet" being used to identify the desired
 // product.
-func Activate(conn connection.Connection, identifier, version, arch string) (*Metadata, *Product, error) {
-	return &Metadata{}, &Product{}, nil
+func Activate(conn connection.Connection, identifier, version, arch, regcode string) (*Metadata, *Product, error) {
```

Comment:
Added `regcode` to support products which require an additional registration code. Did you not added specific tests thoug. Fine?

Reply:
Fine by me!